src/awsS3Module.ts

```
@@ -0,0 +1,65 @@
+import * as AWS from "aws-sdk";
+import * as fs from "fs";
+import * as winston from "winston";
+import { createDirIfNotExists, getDirectoriesPathArray } from "./filesHelper";
+
+export class AwsS3Module {
+    private s3Client: AWS.S3;
+    private bucketName: string;
+    private tempFolder: string;
+
+    constructor(
+        awsAccessId: string,
+        awsSecretKey: string,
+        awsRegion: string,
+        awsBucketName: string,
+        tempFolder: string) {
+        this.bucketName = awsBucketName;
+        this.tempFolder = tempFolder;
+        try {
+            const awsConfig = { accessKeyId: awsAccessId, secretAccessKey: awsSecretKey, region: awsRegion };
+            this.s3Client = new AWS.S3(awsConfig);
+        } catch (ex) {
+            winston.log("error initializing s3 client: " + ex);
```

I think you should throw an error here